server/operator.go

```diff
@@ -160,9 +255,28 @@ func (op *regionOperator) GetResourceKind() ResourceKind {
 	return op.Kind
 }
 
+func (op *regionOperator) GetState() OperatorState {
+	return op.State
+}
+
+func (op *regionOperator) SetState(state OperatorState) {
+	if op.State == OperatorFinished || op.State == OperatorTimeOut {
```

Comment:
can we meet the case that operator is finished or timeout, but the state is still set?